tests/lib/nested.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+. $TESTSLIB/systemd.sh
+
+wait_for_ssh(){
+    retry=300
+    while ! execute_remote true; do
+        retry=$(( retry - 1 ))
+        if [ $retry -le 0 ]; then
+            echo "Timed out waiting for ssh. Aborting!"
+            exit 1
+        fi
+        sleep 1
+    done
+}
+
+prepare_ssh(){
+    execute_remote "sudo adduser --extrausers --quiet --disabled-password --gecos '' test"
+    execute_remote "echo test:ubuntu | sudo chpasswd"
+    execute_remote "echo 'test ALL=(ALL) NOPASSWD:ALL' | sudo tee /etc/sudoers.d/test-user"
+}
+
+create_assertions_disk(){
+    dd if=/dev/null of=assertions.disk bs=1M seek=1
+    mkfs.ext4 -F assertions.disk
+    mkdir /mnt/assertions
+    mount -t ext4 -o loop assertions.disk /mnt/assertions
+    cp $TESTSLIB/assertions/auto-import.assert /mnt/assertions
+    umount /mnt/assertions && rm -rf /mnt/assertions
+}
+
+create_nested_core_vm(){
+    # determine arch related vars
+    case "$NESTED_ARCH" in
+    amd64)
+        QEMU="$(which qemu-system-x86_64)"
+        ;;
+    i386)
+        QEMU="$(which qemu-system-i386)"
+        ;;
+    *)
+        echo "unsupported architecture"
+        exit 1
+        ;;
+    esac
+
+    # create ubuntu-core image
+    mkdir -p /tmp/work-dir
+    snap install --devmode --beta ubuntu-image
+    trap 'snap remove ubuntu-image' EXIT
```

Comment:
Trap will replace other existing traps on the same signal. So if we alread use a `trap ... EXIT` anwhere this is overriden now. We may consider switching to a helper (`add_trap`) that stacks the cleanups if we already use traps. Probably material for a new PR though.